TypeCobol/Compiler/Parser/ProgramClassDocument.cs

```diff
@@ -34,7 +34,7 @@ public ProgramClassDocument(TemporarySemanticDocument previousStepSnapshot, int
         /// <summary>
         /// Snapshot of the code elements document which was used to compute the current step
         /// </summary>
-        public IDocumentSnapshot<ICodeElementsLine> PreviousStepSnapshot { get; private set; }
+        public TemporarySemanticDocument PreviousStepSnapshot { get; private set; }
```

Comment:
I think other `PreviousStepSnapshot` properties can in TemporarySemanticDocument , CodeElementsDocument, ... can use the most derived type to avoid cast in CompilationUnit.

Reply:
With current code no, because this property is defined by `INextStepDocumentSnapshot<T>` as an `IDocumentSnapshot<T>`. C# does not allow (yet) override of inherited method return types. As for `ProgramClassDocument`, it does not implement `INextStepDocumentSnapshot` so I was ablet to change property type without impact.

Reply:
Ok.